src/rpc.rs

```diff
@@ -74,26 +95,31 @@ impl Request {
         buff.read_exact(&mut amino_pre)?;
         buff.set_position(0);
         // TODO: find a way to get back the buffer without cloning the cursor here:
-        let rem: Vec<u8> = buff.clone().into_inner();
+        let rem: Vec<u8> =
+            buff.clone().into_inner()[..(len.checked_add(1).unwrap() as usize)].to_vec();
         if amino_pre == *PP_PREFIX {
```

Comment:
I guess I missed this in previous review, but all of these `if amino_pre`s look like a good case for a `match` expression

Reply:
The match wouldn't be simply
```rust
match amino_pre {
    PP_PREFIX => return Ok(Request::PoisonPill(PoisonPillMsg {})),
    ...
``` 
but something slightly more clumsy like
```rust
match amino_pre {
    ref pp if *pp == *PP_PREFIX => return Ok(Request::PoisonPill(PoisonPillMsg {})),
    ...
``` 
but I guess this is still more idiomatic.

Reply:
Why can't you do:

```rust
match amino_pre.as_ref() {
     PP_PREFIX => Ok(Request::PoisonPill(PoisonPillMsg {})),
     ...
```

...or thereabouts?

Reply:
I tried this (and variants of this), it yields:
```
- a static `PP_PREFIX` is defined here
...
95 |               PP_PREFIX => Ok(Request::PoisonPill(PoisonPillMsg {})),
   |               ^^^^^^^^^ cannot be named the same as a static
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0530]: match bindings cannot shadow statics
  --> src/rpc.rs:96:13
```